### PR TITLE
[4.0] Add prepared statements to plugin privacy consents

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
Use the plugin in all ways you can think of.

* Create users
* Generate privacy consents entries.
* Create a export request.

### Expected result
Nothing changed.

### CODE REVIEW

File: plugins/privacy/consents/consents.php [hunk 1]
```diff
@@ -7,6 +7,8 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use Joomla\Database\ParameterType;
```
Comment (Quy) [inline plugins/privacy/consents/consents.php]: Move after `defined`.

File: plugins/privacy/consents/consents.php [hunk 2]
```diff
@@ -35,13 +37,15 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
...
 		$query = $this->db->getQuery(true)
-			->select('*')
-			->from($this->db->quoteName('#__privacy_consents'))
-			->where($this->db->quoteName('user_id') . ' = ' . (int) $user->id)
-			->order($this->db->quoteName('created') . ' ASC');
+			->select($db->quoteName('*'))
+			->from($db->quoteName('#__privacy_consents'))
+			->where($db->quoteName('user_id') . ' = :id')
+			->order($db->quoteName('created') . ' ASC')
+			->bind(':id', (int) $user->id, ParameterType::INTEGER);
```
Comment (Quy) [inline plugins/privacy/consents/consents.php]: Just checking to make sure it is not an issue as reported here with casting: https://github.com/joomla/joomla-cms/pull/25051#pullrequestreview-253112469

File: plugins/privacy/consents/consents.php [hunk 3]
```diff
@@ -35,13 +37,15 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
...
-		$domain    = $this->createDomain('consents', 'joomla_consent_data');
+		$domain = $this->createDomain('consents', 'joomla_consent_data');
+		$db     = $this->db;
 
 		$query = $this->db->getQuery(true)
-			->select('*')
-			->from($this->db->quoteName('#__privacy_consents'))
-			->where($this->db->quoteName('user_id') . ' = ' . (int) $user->id)
-			->order($this->db->quoteName('created') . ' ASC');
+			->select($db->quoteName('*'))
```
Comment (SharkyKZ) [inline plugins/privacy/consents/consents.php]:
```suggestion
			->select('*')
```

File: plugins/privacy/consents/consents.php (hunk 2 above)
Comment (SharkyKZ) [reply plugins/privacy/consents/consents.php]: It is.

File: plugins/privacy/consents/consents.php [hunk 4]
```diff
@@ -35,13 +37,15 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
 			return array();
 		}
 
-		$domain    = $this->createDomain('consents', 'joomla_consent_data');
+		$domain = $this->createDomain('consents', 'joomla_consent_data');
+		$db     = $this->db;
 
 		$query = $this->db->getQuery(true)
```
Comment (Quy) [inline plugins/privacy/consents/consents.php]:
```suggestion
		$query = $db->getQuery(true)
```

File: plugins/privacy/consents/consents.php [hunk 5]
```diff
@@ -35,13 +37,15 @@ public function onPrivacyExportRequest(PrivacyTableRequest $request, JUser $user
...
 			->select('*')
-			->from($this->db->quoteName('#__privacy_consents'))
-			->where($this->db->quoteName('user_id') . ' = ' . (int) $user->id)
-			->order($this->db->quoteName('created') . ' ASC');
+			->from($db->quoteName('#__privacy_consents'))
+			->where($db->quoteName('user_id') . ' = :id')
+			->order($db->quoteName('created') . ' ASC')
+			->bind(':id', $user->id, ParameterType::INTEGER);
 
 		$items = $this->db->setQuery($query)->loadAssocList();
```
Comment (Quy) [inline plugins/privacy/consents/consents.php]:
```suggestion
		$items = $db->setQuery($query)->loadAssocList();
```

**Comment by SharkyKZ** [comment]
I have tested this item :white_check_mark: successfully on 84afe576958fb62714775fd3d797f4131fb23a43<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25410">issues.joomla.org/tracker/joomla-cms/25410</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 84afe576958fb62714775fd3d797f4131fb23a43<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25410">issues.joomla.org/tracker/joomla-cms/25410</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25410">issues.joomla.org/tracker/joomla-cms/25410</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**